engine/src/mbliphonemail.mm

```diff
@@ -654,5 +654,9 @@ void MCSystemGetCanSendMail(bool& r_result)
 
 void MCSystemMailResult(MCStringRef& r_result)
 {
-	r_result = MCValueRetain(kMCEmptyString);
+	// PM-2016-02-29: [[ Bug 17031 ]] Make sure we return 'the result'
+	if (MCValueGetTypeCode(MCresult -> getvalueref()) == kMCValueTypeCodeString)
+		r_result = MCValueRetain((MCStringRef)MCresult -> getvalueref());
```

Comment:
For the future, sometimes the operator precedence can be clearer by using the C++ constructor-style type cast, e.g.:

```C++
r_result = MCValueRetain(MCStringRef(MCresult->getvalueref()));
```